samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -206,9 +243,174 @@ def _translate_to_cfn(tf_json: dict) -> dict:
+        CloudFormation resources
+    output_directory_path: str
+        the output directory path to write the generated metadata and makefile
+    terraform_application_dir: str
+        the terraform project root directory
+    """
+
+    def _enrich_zip_lambda_function(sam_metadata_resource: Dict, cfn_resource: Dict, cfn_resource_logical_id: str):
+        pass
+
+    def _enrich_image_lambda_function(sam_metadata_resource: Dict, cfn_resource: Dict, cfn_resource_logical_id: str):
+        pass
+
+    resources_types_enrichment_functions = {
+        "ZIP_LAMBDA_FUNCTION": _enrich_zip_lambda_function,
+        "IMAGE_LAMBDA_FUNCTION": _enrich_image_lambda_function,
+    }
+
+    for sam_metadata_resource in sam_metadata_resources:
+        resource_type = (
+            sam_metadata_resource.sam_metadata_resource.get("values", {}).get("triggers", {}).get("resource_type")
+        )
+        sam_metadata_resource_address = sam_metadata_resource.sam_metadata_resource.get("address")
+        enrichment_function = resources_types_enrichment_functions.get(resource_type)
+        if not enrichment_function:
+            raise InvalidSamMetadataProperties(
+                f"The resource type {resource_type} found in the sam metadata resource {sam_metadata_resource_address} "
+                f"is not a correct resource type. The resource type should be one of these values "
+                f"{resources_types_enrichment_functions.keys()}"
+            )
+
+
+def _get_relevant_cfn_resource(
+    sam_metadata_resource: SamMetadataResource, cfn_resources: Dict[str, Dict]
+) -> Tuple[Dict, str]:
+    """
+    use the sam metadata resource name property to determine the resource address, and transform the address to logical
+    id to use it to get the cfn_resource.
+
+    Parameters
+    ----------
+    sam_metadata_resource: SamMetadataResource
+        sam metadata resource that contain extra information about some resource.
+    cfn_resources: Dict
+        CloudFormation resources
+
+    Returns
+    -------
+    tuple(Dict, str)
+        The cfn resource that mentioned in the sam metadata resource, and the resource logical id
+    """
+    sam_metadata_resource_address = sam_metadata_resource.sam_metadata_resource.get("address")
+    resource_name = (
+        sam_metadata_resource.sam_metadata_resource.get("values", {}).get("triggers", {}).get("resource_name")
+    )
+    if not resource_name:
+        raise InvalidSamMetadataProperties(
+            f"sam cli expects the sam metadata resource {sam_metadata_resource_address} to contain a resource name "
+            f"that will be enriched using this metadata resource"
+        )
+
+    # the provided resource name can be the name without the module address, or can be a complete resource address name
+    # check first if the provided name is without the module address
+    LOG.info(
+        "Check if the input source name %s is a postfix to the current module address %s",
```

Comment:
I think let's change `source` to `resource`, to be consistent with the other info messages here.

Reply:
Done